src/config.js

```diff
@@ -0,0 +1,312 @@
+  }
+};
+
+/**
+ * Reads and parses a jsonr-config.json file
+ * Returns the config object or null if file doesn't exist or is invalid
+ */
+const readConfigFile = async (filePath) => {
+  try {
+    const config = await readJsonFile(filePath);
+    logger.debug(`Successfully loaded config from ${filePath}`);
+    return config;
+  } catch (err) {
+    if (err instanceof Deno.errors.NotFound) {
+      logger.debug(`Config file not found at ${filePath}`);
+      return null;
+    }
+    logger.debug(`Error reading config file ${filePath}: ${err}`);
+    logger.error(
+      `Warning: Failed to parse ${filePath}. Ensure it contains valid JSON.`,
+    );
+    return null;
+  }
+};
+
+/**
+ * Loads secrets from a secrets file
+ * Returns an object with variables from the secrets file
+ */
+const loadSecretsFile = async (secretsPath) => {
+  if (!secretsPath) {
+    return {};
+  }
+
+  logger.debug(`Loading secrets from ${secretsPath}`);
+
+  try {
+    const secrets = await readJsonFile(secretsPath);
+    logger.debug(`Loaded ${Object.keys(secrets).length} variables from secrets file`);
+    return secrets;
+  } catch (err) {
+    if (err instanceof Deno.errors.NotFound) {
+      logger.error(
+        `Error: Secrets file not found at ${secretsPath}. Please check the path in your config file.`,
+      );
+      Deno.exit(1);
+    }
+    logger.error(
+      `Error: Failed to read secrets file at ${secretsPath}: ${err.message}`,
+    );
+    Deno.exit(1);
+  }
+};
+
+/**
+ * Finds all jsonr-config.json files from the current directory up to the home directory
+ * Returns an array of config objects, ordered from home directory to current directory
+ * (so that configs closer to cwd can override parent configs)
+ */
+const findConfigFiles = async () => {
+  const configs = [];
+  const homeDir = Deno.env.get("HOME") || Deno.env.get("USERPROFILE");
+
+  if (!homeDir) {
+    logger.debug("Could not determine home directory");
+    return configs;
+  }
+
+  let currentDir = Deno.cwd();
+  const homeDirResolved = deps.resolve(homeDir);
+
+  logger.debug(
+    `Searching for ${CONFIG_FILE_NAME} files from ${currentDir} to ${homeDirResolved}`,
+  );
+
+  // Collect all directories from cwd to home
+  const directories = [];
+  while (true) {
+    directories.push(currentDir);
+
+    // Stop if we've reached the home directory
+    if (currentDir === homeDirResolved) {
+      break;
+    }
+
+    // Stop if we've reached the root directory
+    const parentDir = deps.dirname(currentDir);
+    if (parentDir === currentDir) {
+      break;
+    }
+
+    currentDir = parentDir;
+  }
+
+  // Read configs in reverse order (from home to cwd)
+  // so that configs array is ordered from least specific to most specific
+  for (let i = directories.length - 1; i >= 0; i--) {
+    const dir = directories[i];
+    const configPath = deps.join(dir, CONFIG_FILE_NAME);
+    const config = await readConfigFile(configPath);
```

Comment:
Add logger debug (if it's not added) that will indicate what directory is currently read